plugins/vpp/vpp.go

```diff
@@ -64,19 +66,25 @@ func FindCompatibleBinapi(ch CompatibilityChecker) (binapi.Version, error) {
 	}{}
 	for version, msgList := range binapi.Versions {
 		msgs := msgList.AllMessages()
-		if err := ch.CheckCompatiblity(msgs...); err == nil {
+		if err := ch.CheckCompatiblity(msgList.Core.AllMessages()...); err == nil {
+			logging.Debugf("binapi version %v core compatible", version)
+		} else {
+			logging.Debugf("binapi version %v core check failed: %v", version, err)
+			continue
+		}
+		if err := ch.CheckCompatiblity(msgList.Plugins.AllMessages()...); err == nil {
 			logging.Debugf("found compatible binapi version: %v", version)
 			return version, nil
 		} else if ierr, ok := err.(*govppapi.CompatibilityError); ok {
 			logging.Debugf("binapi version %-15v incompatible: %d/%d incompatible messages",
 				version, len(ierr.IncompatibleMessages), len(msgs))
-
-			if mostCompatible.version == "" || mostCompatible.incompatible > len(ierr.IncompatibleMessages) {
+			incompatible := len(ierr.IncompatibleMessages)
+			if (mostCompatible.version == "" || mostCompatible.incompatible > incompatible) && incompatible < len(msgs) {
```

Comment:
`incompatible` will be always less than length of all messages, because here we check for compatibility only `msgList.Plugins.AllMessages()`